src/gtc/dace/expansion/daceir_builder.py

```diff
@@ -103,6 +105,38 @@ def _get_tasklet_inout_memlets(node: oir.HorizontalExecution, *, get_outputs, gl
     return res
 
 
+def _all_stmts_same_region(scope_nodes, axis: dcir.Axis, interval):
+    return (
+        axis in dcir.Axis.dims_horizontal()
+        and isinstance(interval, dcir.DomainInterval)
+        and all(
+            isinstance(stmt, oir.MaskStmt) and isinstance(stmt.mask, common.HorizontalMask)
+            for tasklet in iter_tree(scope_nodes).if_isinstance(dcir.Tasklet)
+            for stmt in tasklet.stmts
+        )
```

Comment:
I like this, but for readability in the future, could you make a variable for each of these conditions with a name that describes it then `return x and y and z`?

Reply:
done